### DISCUSSION TITLE
Model with Concatenation Layer

### DISCUSSION THREAD

**Discussion by williamliao28** [Xilinx/finn#825] [Q&A] [answered]
Hello,

I have a model containing a concatenation layer. I found that the output quantization of the concatenation layer always has `finn_datatype: FLOAT32`. This is causing problem for the layer after concatenation because the input of that layer will have `finn_datatype: FLOAT32` and will not be converted to HLS layers. Is there a way to make the output quantization to have `finn_datatype: INT`?

<img width="487" alt="截圖 2023-05-26 14 30 49" src="https://github.com/Xilinx/finn/assets/20777211/0d2e65af-a4d9-4a14-96de-fe702fa6c759">

**Comment by MFJI** [comment]
Is it possible to use a QuantIdentity layer after the concat layer?

**Comment by auphelia** [comment, marked as answer]
This problem is solved by PR https://github.com/Xilinx/finn/pull/857 and was discussed in more detail in Issue https://github.com/Xilinx/finn/issues/853.
